bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,760 @@
+  /**
+   * the maxValue of this bsi
+   */
+  private long maxValue;
+
+  /**
+   * the minValue of this bsi
+   */
+  private long minValue;
+
+  /**
+   * the bit component slice Array of this bsi
+   */
+  private Roaring64NavigableMap[] bA;
+
+  /**
+   * the exist bitmap of this bsi which means the columnId have value in this bsi
+   */
+  private Roaring64NavigableMap ebM;
+
+  private Boolean runOptimized = false;
+
+  /*
+   * The default serialization format is {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE}.
+   * all underlying 'Roaring64NavigableMap's should have the same serialization format
+   */
+  private int serializationMode = SERIALIZATION_MODE_PORTABLE;
+
+  /**
+   * NewDefaultBSI constructs an auto-sized BSI, with portable serialization format
+   * for underlying Roaring64NavigableMap as default.
+   */
+  public Roaring64NavigableMapSliceIndex() {
+    this(0L, 0L, SERIALIZATION_MODE_PORTABLE);
+  }
+
+  /**
+   * NewBSI constructs a new BSI.  Min/Max values are optional.  If set to 0
+   * then the underlying BSI will be automatically sized. with portable serialization format
+   * for underlying Roaring64NavigableMap as default.
+   */
+  public Roaring64NavigableMapSliceIndex(long minValue, long maxValue) {
+    this(minValue, maxValue, SERIALIZATION_MODE_PORTABLE);
+  }
+
+  /**
+   * NewBSI constructs a new BSI.  Min/Max values are optional.  If set to 0
+   * then the underlying BSI will be automatically sized.
+   * The serializationMode can be specified as 0(legacy)/1(portable) for underlying Roaring64NavigableMap.
+   */
+  public Roaring64NavigableMapSliceIndex(long minValue, long maxValue, int serializationMode) {
+    if (minValue < 0) {
+      throw new IllegalArgumentException("Values should be non-negative");
+    }
+    if (maxValue < minValue) {
+      throw new IllegalArgumentException("maxValue should GE minValue");
+    }
+    if (serializationMode != SERIALIZATION_MODE_PORTABLE
+        && serializationMode != SERIALIZATION_MODE_LEGACY) {
+      // invalid serialization mode
+      throw new IllegalArgumentException("Invalid serialization mode");
+    }
+
+    this.bA = new Roaring64NavigableMap[64 - Long.numberOfLeadingZeros(maxValue)];
+    for (int i = 0; i < bA.length; i++) {
+      this.bA[i] = new Roaring64NavigableMap();
+    }
+
+    this.ebM = new Roaring64NavigableMap();
+    setSerializationMode(serializationMode);
+  }
+
+  /**
+   * set the serialization mode for underlying Roaring64NavigableMap
+   *
+   * @param serializationMode valid values: SERIALIZATION_MODE_PORTABLE or SERIALIZATION_MODE_LEGACY
+   */
+  public void setSerializationMode(int serializationMode) {
+    if (serializationMode != SERIALIZATION_MODE_PORTABLE
+        && serializationMode != SERIALIZATION_MODE_LEGACY) {
+      // invalid serialization mode
+      throw new IllegalArgumentException("Invalid serialization mode");
+    }
+    this.serializationMode = serializationMode;
```

Comment:
I still feel weird having a BSI serialziation mode, essentially used to change the underlying bitmap serilization mode.